nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/AbstractAuditStorageTest.java

```diff
@@ -0,0 +1,171 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gethin James
+ */
+package org.nuxeo.ecm.platform.audit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_DOC_PATH;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_EVENT_ID;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_ID;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.ScrollResult;
+import org.nuxeo.ecm.core.query.sql.model.Literals;
+import org.nuxeo.ecm.core.query.sql.model.Operator;
+import org.nuxeo.ecm.core.query.sql.model.Predicate;
+import org.nuxeo.ecm.core.query.sql.model.Reference;
+import org.nuxeo.ecm.platform.audit.api.AuditQueryBuilder;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.audit.api.Predicates;
+import org.nuxeo.ecm.platform.audit.impl.LogEntryImpl;
+import org.nuxeo.ecm.platform.audit.service.AbstractAuditBackend;
+import org.nuxeo.ecm.platform.audit.service.AuditBackend;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Audit test that are common across all storage backends
+ */
+public abstract class AbstractAuditStorageTest {
+
+    public static final String ID_FOR_AUDIT_STORAGE_TESTS = "ID_FOR_AUDIT_STORAGE_TESTS";
+
+    public static final int NUM_OF_EVENTS = 56;
+
+    @Inject
+    protected AuditBackend auditBackend;
+
+    protected static ObjectMapper mapper = new ObjectMapper();
+
+    /**
+     * Sets up data for the tests below. Doesn't run using @Before so as not to interfere with other audit backends
+     */
+    public void setUpTestData() throws Exception {
+
+        AbstractAuditBackend backend = (AbstractAuditBackend) this.auditBackend;
+        AuditQueryBuilder builder = new AuditQueryBuilder().predicates(
+                Predicates.eq(LOG_EVENT_ID, ID_FOR_AUDIT_STORAGE_TESTS));
+        List<LogEntry> logs = backend.queryLogs(builder);
+
+        if (logs.isEmpty()) {
+
+            List<String> jsonEntries = new ArrayList<>();
+
+            for (int i = 1; i <= NUM_OF_EVENTS; i++) {
+                ObjectNode logEntryJson = mapper.createObjectNode();
+                logEntryJson.put(LOG_ID, 1000L+Integer.valueOf(i).longValue());
```

Comment:
Can you format the class ?
- space between `1000L` and `+`
- no new line between `jsonEntries` and the `for` loop.